builtin/providers/aws/resource_aws_autoscaling_group.go

```diff
@@ -121,32 +123,27 @@ func resourceAwsAutoscalingGroup() *schema.Resource {
 }
 
 func resourceAwsAutoscalingGroupCreate(d *schema.ResourceData, meta interface{}) error {
-	autoscalingconn := meta.(*AWSClient).autoscalingconn
-
-	var autoScalingGroupOpts autoscaling.CreateAutoScalingGroup
-	autoScalingGroupOpts.Name = d.Get("name").(string)
-	autoScalingGroupOpts.HealthCheckType = d.Get("health_check_type").(string)
-	autoScalingGroupOpts.LaunchConfigurationName = d.Get("launch_configuration").(string)
-	autoScalingGroupOpts.MinSize = d.Get("min_size").(int)
-	autoScalingGroupOpts.MaxSize = d.Get("max_size").(int)
-	autoScalingGroupOpts.SetMinSize = true
-	autoScalingGroupOpts.SetMaxSize = true
-	autoScalingGroupOpts.AvailZone = expandStringList(
+	autoscalingconn := meta.(*AWSClient).awsAutoScalingconn
+
+	var autoScalingGroupOpts autoscaling.CreateAutoScalingGroupType
+	autoScalingGroupOpts.AutoScalingGroupName = aws.String(d.Get("name").(string))
+	autoScalingGroupOpts.HealthCheckType = aws.String(d.Get("health_check_type").(string))
+	autoScalingGroupOpts.LaunchConfigurationName = aws.String(d.Get("launch_configuration").(string))
+	autoScalingGroupOpts.MinSize = aws.Integer(d.Get("min_size").(int))
```

Comment:
min_size and max_size are required, and `Get` will always return a zero value if not set ("0"), so this should be fine.